### ISSUE TITLE
Role of Sweeps object

### ISSUE THREAD

**Issue by emstoudenmire** [mtfishman/ITensorTDVP.jl#11]
Right now, the `Sweeps` type plays a pretty central role in the implementation of `tdvp` and `ITensorTDVP.dmrg` internally. Externally, at an interface level, it's now less central and can be replaced by keyword arguments such as `maxdim=[10,20,100]` and similar. 

A point to discuss is whether we need the `Sweeps` type at all, except for backward compatibility? A key thing it offers are some functions that pad out variables like `maxdim` to have `nsweeps` entries, but this could be done now inside the `_tdvp_compute_sweeps` function i.e. it could process those keyword args and pad them out, replacing them with new values and returning a keyword arg named tuple.

**Comment by mtfishman** [comment]
Agreed, I think we'll find that people will mostly use the keyword arguments interface (I definitely prefer it now over using the Sweeps object), and we can just handle the padding logic with a standard keyword argument/NamedTuple processing function.

We can keep the Sweeps object for backwards compatibility and extract the keyword arguments from the Sweeps object.

One more sophisticated thing it does is the string table syntax:
https://github.com/ITensor/ITensors.jl/blob/ba5a279a6112b1366a44739385b9f435ed3126fc/examples/dmrg/input_files/script/input.jl#L8-L16

but that could be handled without the `Sweeps` object by passing it to a `sweeps` keyword argument which then returns a `NamedTuple` of `maxdim`, `cutoff`, etc.

**Comment by emstoudenmire** [comment]
This issue has now been addressed by recent redesigns of the package, so I'll close it.
